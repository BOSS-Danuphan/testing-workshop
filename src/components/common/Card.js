import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import * as COLORS from '../../constants/colors'

export const Container = styled.div`
  padding: 10px;
  border-bottom: 1px solid ${COLORS.GREY_200};
  background-color: ${COLORS.WHITE};
`
Container.displayName = 'Container'

export const Text = styled.p`
  color: ${COLORS.BLACK};
  font-size: 14px;
`
Text.displayName = 'Text'

export const Title = Text.extend`
  font-weight: bold;
`
Title.displayName = 'Title'

const Card = ({ id, title, body }) => (
  <Container>
    <Title id={`card-title-${id}`}>
      {title}
    </Title>
    <Text id={`card-body-${id}`}>
      {body}
    </Text>
  </Container>
)

Card.defaultProps = {
  title: 'Anonymous',
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  body: PropTypes.string.isRequired,
}

export default Card
