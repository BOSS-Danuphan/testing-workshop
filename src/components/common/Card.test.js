import React from 'react'
import { shallow } from 'enzyme'
import Card, { Container, Text, Title } from './Card'

describe('Test card ui structure', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      id: 1,
      title: 'Lorem',
      body: 'Ipsum.',
    }

    wrapper = shallow(<Card {...props} />)
  })

  it('Should render correct card title', () => {
    const cardTitle = wrapper.find('#card-title-1')

    expect(cardTitle).toHaveLength(1)
    expect(cardTitle.props()).toHaveProperty('children', 'Lorem')
  })

  it('Should render correct card body', () => {
    const cardBody = wrapper.find('#card-body-1')

    expect(cardBody).toHaveLength(1)
    expect(cardBody.props()).toHaveProperty('children', 'Ipsum.')
  })

  describe('Match its snapshot', () => {
    it('When all props exists', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('When no props exists', () => {
      props.title = undefined
      wrapper = shallow(<Card {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})

describe('Test Container styled-components', () => {
  it('Match its snapshot', () => {
    const wrapper = shallow(<Container />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('Test Text styled-components', () => {
  it('Match its snapshot', () => {
    const wrapper = shallow(<Text />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('Test Title styled-components', () => {
  it('Match its snapshot', () => {
    const wrapper = shallow(<Title />)

    expect(wrapper).toMatchSnapshot()
  })
})
