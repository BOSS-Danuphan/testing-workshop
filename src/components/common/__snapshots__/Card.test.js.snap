// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Container styled-components Match its snapshot 1`] = `
.c0 {
  padding: 10px;
  border-bottom: 1px solid #E6ECF0;
  background-color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`Test Text styled-components Match its snapshot 1`] = `
.c0 {
  color: #14171A;
  font-size: 14px;
}

<p
  className="c0"
/>
`;

exports[`Test Title styled-components Match its snapshot 1`] = `
.c0 {
  color: #14171A;
  font-size: 14px;
  font-weight: bold;
}

<p
  className="c0"
/>
`;

exports[`Test card ui structure Match its snapshot When all props exists 1`] = `
<Container>
  <Title
    id="card-title-1"
  >
    Lorem
  </Title>
  <Text
    id="card-body-1"
  >
    Ipsum.
  </Text>
</Container>
`;

exports[`Test card ui structure Match its snapshot When no props exists 1`] = `
<Container>
  <Title
    id="card-title-1"
  >
    Anonymous
  </Title>
  <Text
    id="card-body-1"
  >
    Ipsum.
  </Text>
</Container>
`;
